pyomo/contrib/iis/mis.py

```diff
@@ -0,0 +1,371 @@
+
+    Neither the name of the University of California, Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory, National Renewable Energy Laboratory, National Energy Technology Laboratory, U.S. Dept. of Energy nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+You are under no obligation whatsoever to provide any bug fixes, patches, or upgrades to the features, functionality or performance of the source code ("Enhancements") to anyone; however, if you choose to make your Enhancements available either publicly, or directly to Lawrence Berkeley National Laboratory, without imposing a separate written license agreement for such Enhancements, then you hereby grant the following license: a non-exclusive, royalty-free perpetual license to install, use, modify, prepare derivative works, incorporate into other computer software, distribute, and sublicense such enhancements or derivative works thereof, in binary and source code form.
+"""
+"""
+Minimal Intractable System (MIS) finder
+Originally written by Ben Knueven as part of the WaterTAP project:
+   https://github.com/watertap-org/watertap
+That's why this file has the watertap copyright notice.
+
+copied by DLW 18Feb2024 and edited
+
+See: https://www.sce.carleton.ca/faculty/chinneck/docs/CPAIOR07InfeasibilityTutorial.pdf
+"""
+
+import logging
+import pyomo.environ as pyo
+
+from pyomo.core.plugins.transform.add_slack_vars import AddSlackVariables
+
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+
+from pyomo.common.modeling import unique_component_name
+from pyomo.common.collections import ComponentMap, ComponentSet
+
+from pyomo.opt import WriterFactory
+
+logger = logging.getLogger("pyomo.contrib.iis")
+logger.setLevel(logging.INFO)
+
+
+class _VariableBoundsAsConstraints(IsomorphicTransformation):
+    """Replace all variables bounds and domain information with constraints.
+
+    Leaves fixed Vars untouched (for now)
+    """
+
+    def _apply_to(self, instance, **kwds):
+
+        boundconstrblockname = unique_component_name(instance, "_variable_bounds")
```

Comment:
Not a reason to hold up this PR, but this name would be pretty inaccessible to someone new or someone who is using a screen reader to parse the code. I suggest adding in some breaks:
```
bound_constr_block_name
```